modules/hal_infineon/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2020 Linumiz
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library_named(modules_hal_infineon)
```

Comment:
This is causing the CMake warning.

`modules_hal_infineon` library is defined even if none of below subdirectories is added; so, when not building for Infineon/Cypress devices, the source list for this library will be empty.

This needs to go into `if(CONFIG_HAS_XMCLIB OR CONFIG_SOC_FAMILY_PSOC6)`.

Another option is to add `zephyr_library_property(ALLOW_EMPTY TRUE)`.

Reply:
Hi @stephanosio , thanks for catching this. I am working on PR with fix.

Reply:
PR: https://github.com/zephyrproject-rtos/zephyr/pull/45671